pkg/kv/dist_sender.go

```diff
@@ -1235,115 +1214,29 @@ func (ds *DistSender) sendToReplicas(
 					propagateError = true
 				}
 
-				log.ErrEventf(ctx, "application error: %s", call.Reply.Error)
-
 				if propagateError {
-					if pending > 0 {
-						// If there are still pending RPCs, try to wait them out.
-						//
-						// Note that ambiguous result errors can arrive from these
-						// in-flight RPCs, so if we've sent any, we better wait
-						// for them, even if we're not going to generate ambiguous
-						// results from them ourselves.
-						timer := time.NewTimer(ds.pendingRPCTimeout)
-						defer timer.Stop()
-					pendingLoop:
-						for pending > 0 {
-							select {
-							case pendingCall := <-done:
-								pending--
-								if err := pendingCall.Err; err != nil {
-									if grpc.Code(err) != codes.Unavailable {
-										ambiguousError = err
-									}
-								} else if pendingCall.Reply.Error == nil {
-									return pendingCall.Reply, nil
-								}
-							case <-timer.C:
-								break pendingLoop
-							}
-						}
-					}
-
 					// The error received is likely not specific to this
 					// replica, so we should return it instead of trying other
-					// replicas. However, if we're trying to commit a
-					// transaction and there are still other RPCs outstanding
-					// or an ambiguous RPC error was already received, we must
-					// return an ambiguous commit error instead of the
-					// returned error.
-					if haveCommit {
-						if pending > 0 || ambiguousError != nil {
-							log.ErrEventf(ctx, "returning ambiguous result (error=%v pending=%d)", ambiguousError, pending)
-							return nil, roachpb.NewAmbiguousResultError(
-								fmt.Sprintf("error=%v pending=%d", ambiguousError, pending))
-						}
-					}
+					// replicas.
```

Comment:
Here, you're going to need to resurrect the `ambiguousError` variable and check `haveCommit` to properly return the `AmbiguousResultError` instead of whatever proximate error is received here.